site/content/how-to/traffic-management/snippets-filters.md

```diff
@@ -0,0 +1,345 @@
+---
+title: "Snippets Filters"
+weight: 800
+toc: true
+docs: "DOCS-000"
+---
+
+Learn how to use Snippets with the `SnippetsFilter` API.
+
+## Overview
+
+Snippets allow Cluster Operators to insert NGINX configuration into different contexts of the
+NGINX configurations that NGINX Gateway Fabric generates.
+
+Snippets are for advanced NGINX users who need more control over the generated NGINX configuration,
+and can be used in cases where Gateway API resources or NGINX extensions don't apply.
+
+## Disadvantages of Snippets
+
+Snippets are configured using the `SnippetsFilter` API, but are disabled by default due to their complexity and security implications.
+
+To use Snippets, set the `nginxGateway.snippetsFilters.enable` command line argument to true.
+
+Snippets have the following disadvantages:
+
+- *Complexity*. Snippets require you to:
+  - Understand NGINX Configuration primitives and implement a correct NGINX configuration.
+  - Understand how NGINX Gateway Fabric generates NGINX configuration so that a Snippet doesn’t interfere with the other features in the configuration.
+- *Decreased robustness*. An incorrect Snippet can invalidate NGINX configuration, causing reload failures. Until the snippet is fixed, it will prevent any new configuration updates, including updates for the other Gateway resources.
+- *Security implications*. Snippets give access to NGINX configuration primitives, which are not validated by NGINX Gateway Fabric. For example, a Snippet can configure NGINX to serve the TLS certificates and keys used for TLS termination for Gateway resources.
+
+{{< note >}} If the NGINX configuration includes an invalid Snippet, NGINX will continue to operate with the last valid configuration. {{< /note >}}
+
+Due to the described disadvantages of Snippets, we recommend exhausting all other means of NGINX configuration through Gateway API resources,
+[first-class policies]({{< relref "overview/custom-policies.md" >}}), and other NGINX extensions before using Snippets.
```

Comment:
I don't like the way I implemented the AC of 

> A recommendation at the top of the guide that guides users to use our first-class policies before using snippets.

However, I can't think of anything different, I'm open to suggestions.

Reply:
I'd shift this paragraph up to the start, and maybe lead with the first class policies.

"We recommend managing NGINX configuration through Gateway API resources, first class policies and other NGINX extensions before using Snippets".

You could even wrap it in a warning note if we're really trying to encourage them to use other methods.

Reply:
I moved it, is it in the right spot?

Reply:
Yep, perfect!